pkg/lint/lintersdb/custom_linters.go

```diff
@@ -115,8 +116,11 @@ func (m *Manager) lookupAnalyzerPlugin(plug *plugin.Plugin) ([]*analysis.Analyze
 		return nil, err
 	}
 
-	m.log.Warnf("plugin: 'AnalyzerPlugin' plugins are deprecated, please use the new plugin signature: " +
-		"https://golangci-lint.run/contributing/new-linters/#create-a-plugin")
+	// TODO(ldez): remove this env var in the next minor version (v1.55.0)
+	if os.Getenv("GOLANGCI_LINT_HIDE_WARNING_ABOUT_PLUGIN_API_DEPRECATION") != "1" {
```

Comment:
Personally I think not forcing just `1` but any set value is more appropriate for environment variables and also more common.

```suggestion
	if _, ok := os. LookupEnv("GOLANGCI_LINT_HIDE_WARNING_ABOUT_PLUGIN_API_DEPRECATION"); !ok {
```

Reply:
I put `1` just because it's not a real option but LookupEnv will work.